service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/storageContainer/CreateAzureStorageContainerStep.java

```diff
@@ -0,0 +1,142 @@
+          e.getValue().getCode(),
+          e);
+
+      return new StepResult(StepStatus.STEP_RESULT_FAILURE_RETRY, e);
+    }
+
+    return StepResult.getStepResultSuccess();
+  }
+
+  /**
+   * Deletes the storage container if the container is available. If the storage container is available
+   * and deletes fails, the failure is considered fatal and must looked into it.
+   *
+   * @param context
+   * @return Step result.
+   * @throws InterruptedException
+   */
+  @Override
+  public StepResult undoStep(FlightContext context) throws InterruptedException {
+    final AzureCloudContext azureCloudContext =
+            context
+                    .getWorkingMap()
+                    .get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class);
+    final StorageManager storageManager = crlService.getStorageManager(azureCloudContext, azureConfig);
+
+    try {
+      storageManager
+              .storageAccounts()
+              .getByResourceGroup(
+                      azureCloudContext.getAzureResourceGroupId(), resource.getStorageAccountName());
+    } catch (ManagementException ex) {
+      logger.warn(
+              "Deletion of the storage container is not required. Parent storage account does not exist. {}",
+              resource.getStorageAccountName());
+      return StepResult.getStepResultSuccess();
+    }
+    final PagedIterable<ListContainerItemInner> existingContainers = storageManager.blobContainers().list(
+            azureCloudContext.getAzureResourceGroupId(), resource.getStorageAccountName()
+    );
+    ListContainerItemInner existingContainer = null;
+    for (ListContainerItemInner item : existingContainers) {
+      if (item.name().equals(resource.getStorageContainerName())) {
+        existingContainer = item;
+        break;
+      }
+    }
+    if (existingContainer == null) {
+      logger.warn(
+              "Deletion of the storage container is not required. Storage container does not exist. {}",
+              resource.getStorageContainerName());
+      return StepResult.getStepResultSuccess();
+    } else {
+      try {
+        logger.warn("Attempting to delete storage container '{}' in account '{}'",
+                resource.getStorageContainerName(),
+                resource.getStorageAccountName()
+        );
+        storageManager.blobContainers().delete(
+                azureCloudContext.getAzureResourceGroupId(),
+                resource.getStorageAccountName(),
+                resource.getStorageContainerName()
+        );
+        logger.warn("Successfully deleted storage container '{}' in account '{}'",
+                resource.getStorageContainerName(),
+                resource.getStorageAccountName()
+        );
+        return StepResult.getStepResultSuccess();
+      } catch (ManagementException ex) {
+        logger.error(
+                "Attempt to delete Azure Storage Container failed on this try: " + resource.getStorageContainerName(),
```

Comment:
❓ It will be helpful for debugging to include the error code from the exception in this logging statement, will we get that by including ex as the last arg here? Or do we need to include ex.getValue().getCode() ?